aws/data_source_aws_ssm_parameter.go

```diff
@@ -76,9 +70,9 @@ func dataAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error {
 		Resource:  fmt.Sprintf("parameter/%s", strings.TrimPrefix(d.Id(), "/")),
 	}
 	d.Set("arn", arn.String())
-	d.Set("name", param.Name)
-	d.Set("type", param.Type)
-	d.Set("value", param.Value)
+	d.Set("name", *param.Name)
```

Comment:
To prevent potential panics, we should allow `d.Set()` to safely perform these dereferences or use the SDK methods, e.g. `aws.StringValue(param.Name)`